src/parsing.jl

```diff
@@ -112,20 +112,25 @@ Base.parse(::Type{T}, pqv::PQValue) where {T} = parse(T, string_view(pqv))
 # allow parsing as a Symbol anything which works as a String
 Base.parse(::Type{Symbol}, pqv::PQValue) = Symbol(string_view(pqv))
 
-function Base.start(pqv::PQValue)
-    sv = string_view(pqv)
-    return (sv, start(sv))
-end
+if VERSION >= v"0.7.0-DEV.5125"
+    Base.iterate(pqv::PQValue) = iterate(string_view(pqv))
+    Base.iterate(pqv::PQValue, i) = iterate(string_view(pqv), i)
```

Comment:
This creates `string_view` each time; it should be passed around as part of the state as in the old iteration functions below. That should only hold for `PQValue`, not the other iterators though.